types/ramda/test/map-tests.ts

```diff
@@ -32,29 +32,26 @@ import * as R from 'ramda';
     }
 
     interface B {
-        a: string;
-        b: string;
-    }
-
-    interface C {
         b: number;
         c: string;
     }
 
-    R.map<A, A>(R.inc, { a: 1, b: 2 });
-    R.map<A, B>(R.toString, { a: 1, b: 2 });
-
-    R.map<A, A>(R.inc)({ a: 1, b: 2 });
-    R.map<A, B>(R.toString)({ a: 1, b: 2 });
+    // $ExpectType Record<"a" | "b", number>
+    R.map(R.inc, { a: 1, b: 2 });
+    // $ExpectType Record<"a" | "b", string>
+    R.map(R.toString, { a: 1, b: 2 });
 
-    type KeyOfUnion<T> = T extends infer U ? keyof U : never;
+    // $ExpectType Record<"a" | "b", number>
+    R.map(R.inc)({ a: 1, b: 2 });
+    // $ExpectType Record<"a" | "b", string>
+    R.map(R.toString)({ a: 1, b: 2 });
 
     /**
      * Typescript implementation of union order is not guaranteed and can
      * change. Therefor using `||` here, which is a feature of $ExpectType
      */
     // $ExpectType Record<"c" | "a" | "b", void> || Record<"a" | "b" | "c", void>
-    R.map<A | C, Record<KeyOfUnion<A | C>, void>>(
+    R.map<A | B, void>(
```

Comment:
this is actually no longer necessary, but I couldn't do `as A | B` because of the linting rules so I did this instead.